src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -153,31 +189,34 @@ private static string BuildTopLevelSelect(SelectExpandClause selectExpandClause)
         }
 
         /// <summary>
-        /// Get the string representation of a select item (that isn't an expandedNavPropSelectItem
+        /// Get the string representation of a Pathselect item (that isn't an expandedNavPropSelectItem
         /// </summary>
-        /// <param name="selectedItem">the select item to translate</param>
-        /// <returns>the string representation of this select item, or null if the select item is an expandedNavPropSelectItem</returns>
-        private static string GetSelectString(SelectItem selectedItem)
+        /// <param name="pathSelectItem">the pathselect item to translate</param>
+        /// <returns>the string representation of this select pathselectitem</returns>
+        private static IList<string> GetSelectStringFromPathSelectItem(PathSelectItem pathSelectItem)
         {
-            WildcardSelectItem wildcardSelect = selectedItem as WildcardSelectItem;
-            NamespaceQualifiedWildcardSelectItem namespaceQualifiedWildcard = selectedItem as NamespaceQualifiedWildcardSelectItem;
-            PathSelectItem pathSelectItem = selectedItem as PathSelectItem;
-
-            if (wildcardSelect != null)
+            IList<string> nextLevelSelectList = null;
+            if (pathSelectItem.SelectAndExpand != null)
             {
-                return "*";
+                nextLevelSelectList = GetCurrentLevelSelectList(pathSelectItem.SelectAndExpand);
             }
-            else if (namespaceQualifiedWildcard != null)
-            {
-                return namespaceQualifiedWildcard.Namespace + ".*";
-            }
-            else if (pathSelectItem != null)
+
+            string selectListItem = String.Join("/", pathSelectItem.SelectedPath.WalkWith(PathSegmentToStringTranslator.Instance).ToArray());

```

Comment:
>slator.Instance).ToArray [](start = 113, length = 24)

Same. Is there an alternate to doing a ToArray?

Reply:
Current .NET version string join takes Ienumerable whereas older versions needs array. Other way is to use StringBuilder, Append and return .ToString(). But as of now not sure whether that would give any perf improvement. We can figure out and if it does can have a custom string join method for all string joins